src/tests/end-to-end/utils/utils.h

```diff
@@ -4,24 +4,189 @@
+template<class MatrixType>
+class TimeSeriesConfigurer
+{
+public:
+    TimeSeriesConfigurer() = default;
+    TimeSeriesConfigurer(MatrixType& matrix) : ts_(&matrix) {}
+    TimeSeriesConfigurer& setColumnCount(unsigned int columnCount);
+    TimeSeriesConfigurer& fillColumnWith(unsigned int column, double value);
+private:
+    MatrixType* ts_ = nullptr;
+};
+
+template<class MatrixType>
+TimeSeriesConfigurer<MatrixType>& TimeSeriesConfigurer<MatrixType>::setColumnCount(unsigned int columnCount)
+{
+    ts_->resize(columnCount, HOURS_PER_YEAR);
+    return *this;
+}
+
+template<class MatrixType>
+TimeSeriesConfigurer<MatrixType>& TimeSeriesConfigurer<MatrixType>::fillColumnWith(unsigned int column, double value)
+{
+    ts_->fillColumn(column, value);
+    return *this;
+}
+
+
+class ThermalClusterConfig
+{
+public:
+    ThermalClusterConfig() = default;
+    ThermalClusterConfig(ThermalCluster* cluster);
+    ThermalClusterConfig& setNominalCapacity(double nominalCapacity);
+    ThermalClusterConfig& setUnitCount(unsigned int unitCount);
+    ThermalClusterConfig& setCosts(double cost);
+    ThermalClusterConfig& setAvailablePowerNumberOfTS(unsigned int columnCount);
+    ThermalClusterConfig& setAvailablePower(unsigned int column, double value);
+
+private:
+    ThermalCluster* cluster_ = nullptr;
+    TimeSeriesConfigurer<Matrix<double>> tsAvailablePowerConfig_;
+};
+
+std::shared_ptr<ThermalCluster> addClusterToArea(Area* area, const std::string& clusterName);
+void addScratchpadToEachArea(Study& study);
+
+// -------------------------------
+// Simulation results retrieval
+// -------------------------------
+class averageResults
+{
+public:
+    averageResults(Variable::R::AllYears::AverageData& averageResults) : averageResults_(averageResults)
+    {}
+
+    double hour(unsigned int hour) { return averageResults_.hourly[hour]; }
+    double day(unsigned int day) { return averageResults_.daily[day]; }
+    double week(unsigned int week) { return averageResults_.weekly[week]; }
+
+private:
+    Variable::R::AllYears::AverageData& averageResults_;
+};
+
+
+class OutputRetriever
+{
+public:
+    OutputRetriever(std::shared_ptr<ISimulation<Economy>>& simulation) : simulation_(simulation) {}
+    averageResults overallCost(Area* area);
+    averageResults load(Area* area);
+    averageResults flow(AreaLink* link);
+    averageResults thermalGeneration(ThermalCluster* cluster);
+
+private:
+    template<class VCard>
+    typename Variable::Storage<VCard>::ResultsType* retrieveAreaResults(Area* area);
+
+    template<class VCard>
+    typename Variable::Storage<VCard>::ResultsType* retrieveLinkResults(AreaLink* link);
+
+    template<class VCard>
+    typename Variable::Storage<VCard>::ResultsType* retrieveResultsForThermalCluster(ThermalCluster* cluster);
 
-float defineYearsWeight(Study::Ptr pStudy, const std::vector<float>& yearsWeight);
+    std::shared_ptr<ISimulation<Economy>>& simulation_;
```

Comment:
Weird to have a reference to smart pointer:
we should have either one or the other.